esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+#### getLogProfile
+    string getLogProfile()
+
+Determines if a log profile has been set. If set, the profile name is returned. If not set, returns an empty string.
+
+In the context of log agents and their variant identifiers, the log profile corresponds to a variant group identifier.
+
+#### getStoredStringValue
+    string getStoredStringValue(value_name)
+
+Determines if a named value has been added to the script context's `store` section. Returns the value if it exists. Returns an empty string if it does not exist.
+
+| Parameter | Required? | Description
+| - | - | -
+|value_name | yes | XPath expression evaluating to the name of a stored string value.
+
+#### logOptionExists
+    boolean logOptionExists(option_name)
+
+Determines if a named log option has been set. Returns *true* if set. Returns *false* if not set.
+
+In the context of log agents and their variant identifiers, a log option generally corresponds to a variant type identifier. It is acceptable to set options that are not agent type identifiers.
+
+| Parameter | Required? | Description
+| - | - | -
+|option_name | yes | XPath expression evaluating to the name of a log option.
+
+#### secureAccessFlags
+    number secureAccessFlags(access-level-1, ..., access-level-n)
+
+Returns a bit-mask containing the internal representations of each given access level.
+
+| Parameter | Required? | Description
+| - | - | -
+| access-level-1 | yes | The text representation of a defined resource access level. One value is required. |
+| access-level-2 ... access-level-n | no | The text representation of a defined resource access level. |
+
+#### storedValueExists
+    boolean storedValueExists(value_name)
+
+Determines if a named value exists in the script context's `store` section. Returns *true* if the name exists. Returns *false* if the name does not exist.
+
+| Parameter | Required? | Description
+| - | - | -
+|value_name | yes | XPath expression evaluating to the name of a stored string value.
+
+#### tokenize
+    node-set tokenize(string)
+    node-set tokenize(string, delimiters)
+
+Breaks an input text buffer into tokens based on delimiters. The tokens are temporarily added to the script context as individual nodes which can be used by script operations that act on node-sets.
+
+| Parameter | Required? | Description
+| - | - | -
+|string | yes | Text buffer possibly containing delimited text to be split into tokens.
+| delimiters | no | Optional set of characters used to split *string* into tokens. Defaults to tab, carriage return, line feed, and space, if omitted. |
+
+#### Example
+
+Given a script operation such as `<variable name="tokens" select="tokenize(\"a,b,c,D\", \",\")"/>`, the script engine generates a *GUID* on the fly each token as a node to the context at `temporaries/guid/token` as shown below. The entry in `temporaries` must be copied or used immediately because it does not persist beyond the lifetime of the command invoking `tokenize()`.
```

Comment:
Yes, you can delete the part about using the temporaries right away.  They aren't cleaned up until the script is done.   The token variable is useable until the end.  But I wouldn't even mention the temporaries at all unless you think it's necessary.  That's more of an implementation detail so that they can access the variable contents (or things can be magically passed to the next xpath function).

Reply:
I've made a much more clear example